src/main/java/com/conveyal/datatools/manager/models/Deployment.java

```diff
@@ -405,23 +422,33 @@ public void dump (File output, boolean includeManifest, boolean includeOsm, bool
                 : null;
     }
 
+    /** Generate router config for deployment as byte array (for writing to file output stream). */
+    public String generateRouterConfigAsString() {
+        if (customRouterConfig != null) return customRouterConfig;
+        return writeToString(this.parentProject().routerConfig);
+    }
+
     /**
      * Get OSM extract from OSM vex server as input stream.
      */
     public static InputStream downloadOsmExtract(Rectangle2D rectangle2D) throws IOException {
-        Bounds bounds = new Bounds(rectangle2D);
-        if (!bounds.areValid()) {
-            throw new IllegalArgumentException(String.format("Provided bounds %s are not valid", bounds.toVexString()));
-        }
-        URL vexUrl = new URL(String.format(Locale.ROOT, "%s/%s.pbf",
-                DataManager.getConfigPropertyAsText("OSM_VEX"),
-                bounds.toVexString()));
+        URL vexUrl = getVexUrl(rectangle2D);
         LOG.info("Getting OSM extract at {}", vexUrl.toString());
         HttpURLConnection conn = (HttpURLConnection) vexUrl.openConnection();
         conn.connect();
         return conn.getInputStream();
     }
 
+    public static URL getVexUrl (Rectangle2D rectangle2D) throws MalformedURLException {
+        Bounds bounds = new Bounds(rectangle2D);
+        if (!bounds.areValid()) {
+            throw new IllegalArgumentException(String.format("Provided bounds %s are not valid", bounds.toVexString()));
+        }
+        return new URL(String.format(Locale.ROOT, "%s/%s.pbf",
+            DataManager.getConfigPropertyAsText("OSM_VEX"),
```

Comment:
Should this method check that this config field is provided/valid?

Reply:
I'm pretty sure a MalformedURLException will get through if it isn't defined.